File: tests/machine/x/c/closure.c
#include <stdio.h>
#include <stdlib.h>

static int _lambda0_n;
int _lambda0(int x) { return x + _lambda0_n; }

int (*makeAdder(int n))(int) { return (_lambda0_n = n, _lambda0); }

int main() {
  __auto_type add10 = makeAdder(10);
  printf("%d\n", add10(7));
  return 0;
}

File: tests/machine/x/scala/closure.scala
object closure {
  def makeAdder(n: Int): (Int) => Int = {
    return (x: Int) => (x).asInstanceOf[Int] + (n).asInstanceOf[Int]
  }
  
  def main(args: Array[String]): Unit = {
    val add10 = makeAdder(10)
    println((add10(7)))
  }
}
